### DISCUSSION TITLE
guide/api/node/

### DISCUSSION THREAD

**Discussion by giscus** [pengzhanbo/vuepress-theme-plume#264] [docs-comment]
# guide/api/node/

https://theme-plume.vuejs.press/guide/api/node/

**Comment by MrHaoH** [comment]
大佬，我项目中用了window、localStorage，为啥打包的时候执行"docs:build": "vuepress build docs --clean-cache --clean-temp",这个命令报错：ReferenceError: localStorage is not defined
感谢大佬的帮助

**Reply by pengzhanbo** [reply]
打包的时候需要执行 SSR 构建，SSR 在 NodeJs 环境中构建是不存在 `window` 对象的，会导致构建失败。最简单的解决方法是使用 [vueuse useLocalStorage](https://vueuse.org/core/useLocalStorage/) 代替 `window.localStorage` ，或者 可以在使用前先判断 `window` 是否存在： `typeof window !== 'undefined'`。
